tst/MetricsApiTest.cpp

```diff
@@ -89,7 +89,117 @@ TEST_F(MetricsApiTest, webRtcIceCandidateGetMetrics)
+    EXPECT_PRED_FORMAT2(testing::IsSubstring, "stun.kinesisvideo.us-west-2.amazonaws.com", rtcIceMetrics.rtcStatsObject.localIceCandidateStats.url);
+    EXPECT_PRED_FORMAT2(testing::IsSubstring, "", rtcIceMetrics.rtcStatsObject.localIceCandidateStats.relayProtocol);
+
+    rtcIceMetrics.requestedTypeOfStats = RTC_STATS_TYPE_REMOTE_CANDIDATE;
+    EXPECT_EQ(STATUS_SUCCESS, rtcPeerConnectionGetMetrics(pRtcPeerConnection, NULL, &rtcIceMetrics));
+    EXPECT_EQ(1111, rtcIceMetrics.rtcStatsObject.remoteIceCandidateStats.port);
+    EXPECT_EQ(3, rtcIceMetrics.rtcStatsObject.remoteIceCandidateStats.priority);
+    EXPECT_PRED_FORMAT2(testing::IsSubstring, "10.1.255.255", rtcIceMetrics.rtcStatsObject.remoteIceCandidateStats.address);
+    EXPECT_PRED_FORMAT2(testing::IsSubstring, "host", rtcIceMetrics.rtcStatsObject.remoteIceCandidateStats.candidateType);
+    EXPECT_PRED_FORMAT2(testing::IsSubstring, "", rtcIceMetrics.rtcStatsObject.remoteIceCandidateStats.protocol);
+
+    EXPECT_EQ(STATUS_SUCCESS, closePeerConnection(pRtcPeerConnection));
+    EXPECT_EQ(STATUS_SUCCESS, freePeerConnection(&pRtcPeerConnection));
+}
+
+#ifdef ENABLE_STATS_CALCULATION_CONTROL
+
+TEST_F(MetricsApiTest, webRtcIceServerGetMetrics_IceStatsControlOn_Disabled)
+{
+    RtcConfiguration configuration;
+    PRtcPeerConnection pRtcPeerConnection;
+    RtcStats rtcIceMetrics;
+    rtcIceMetrics.requestedTypeOfStats = RTC_STATS_TYPE_ICE_SERVER; // Supplying a type that is unavailable
+    rtcIceMetrics.rtcStatsObject.iceServerStats.iceServerIndex = 5;
+
+    MEMSET(&configuration, 0x00, SIZEOF(RtcConfiguration));
+    MEMSET(&rtcIceMetrics.rtcStatsObject.iceServerStats, 0x00, SIZEOF(RtcIceServerStats));
+    STRNCPY(configuration.iceServers[0].urls, (PCHAR) "stun:stun.kinesisvideo.us-west-2.amazonaws.com:443", MAX_ICE_CONFIG_URI_LEN);
+    STRNCPY(configuration.iceServers[0].credential, (PCHAR) "", MAX_ICE_CONFIG_CREDENTIAL_LEN);
+    STRNCPY(configuration.iceServers[0].username, (PCHAR) "", MAX_ICE_CONFIG_USER_NAME_LEN);
 
+    STRNCPY(configuration.iceServers[1].urls, (PCHAR) "turns:54.202.170.151:443?transport=tcp", MAX_ICE_CONFIG_URI_LEN);
+    STRNCPY(configuration.iceServers[1].credential, (PCHAR) "username", MAX_ICE_CONFIG_CREDENTIAL_LEN);
+    STRNCPY(configuration.iceServers[1].username, (PCHAR) "password", MAX_ICE_CONFIG_USER_NAME_LEN);
+    configuration.kvsRtcConfiguration.enableIceStats = FALSE;
+    ASSERT_EQ(STATUS_SUCCESS, createPeerConnection(&configuration, &pRtcPeerConnection));
+
+    rtcIceMetrics.rtcStatsObject.iceServerStats.iceServerIndex = 1;
+    EXPECT_EQ(STATUS_SUCCESS, rtcPeerConnectionGetMetrics(pRtcPeerConnection, NULL, &rtcIceMetrics));
+    EXPECT_EQ(0, rtcIceMetrics.rtcStatsObject.iceServerStats.port);
+    EXPECT_EQ('\0', rtcIceMetrics.rtcStatsObject.iceServerStats.url[0]);
+    EXPECT_EQ('\0', rtcIceMetrics.rtcStatsObject.iceServerStats.protocol[0]);
```

Comment:
I would think this is a bit misleading because it seems like WebRTC is not sending anything. Even though it would be in this case just no stats are calculated.